pkg/controller/servicemesh/webhooks/validation/controlplane_upgrade_v1_0.go

```diff
@@ -0,0 +1,117 @@
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "edge"},
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "metric"},
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "quota"},
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "reportnothing"},
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "tracespan"},
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "cloudwatch"},
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "dogstatsd"},
+		schema.GroupVersionKind{Group: "config.istio.io", Version: "v1alpha2", Kind: "zipkin"},
+	}
+)
+
+func (v *ControlPlaneValidator) validateUpgradeFromV1_0(ctx context.Context, smcp *maistrav1.ServiceMeshControlPlane) error {
+	var allErrors []error
+
+	meshNamespaces := sets.NewString(smcp.GetNamespace())
+	smmr, err := v.getSMMR(smcp)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return pkgerrors.Wrap(err, "error retrieving ServiceMeshMemberRoll for mesh")
+		}
+	}
+	meshNamespaces.Insert(smmr.Status.ConfiguredMembers...)
+
+	// return error if any deprecated mixer resources are being used
+	for _, gvk := range unsupportedOldResourcesV1_1 {
+		objects := &unstructured.UnstructuredList{}
+		objects.SetGroupVersionKind(gvk)
+		// XXX: do we list all in the cluster, or list for each member namespace?
+		if err := v.client.List(ctx, nil, objects); err != nil {
+			if !meta.IsNoMatchError(err) && !errors.IsNotFound(err) {
+				return pkgerrors.Wrapf(err, "error listing %s resources", gvk.String())
+			}
+		}
+		objects.EachListItem(func(obj runtime.Object) error {
+			metaObj, err := meta.Accessor(obj)
+			if err != nil {
+				allErrors = append(allErrors, pkgerrors.Wrapf(err, "error accessing object metadata for %s resource", gvk.String()))
+			}
+			// we only care about resources in this mesh, which aren't being managed by the operator directly
+			if meshNamespaces.Has(metaObj.GetNamespace()) && !metav1.IsControlledBy(metaObj, smcp) {
+				allErrors = append(allErrors, fmt.Errorf("%s/%s of type %s is not supported in newer version", metaObj.GetNamespace(), metaObj.GetName(), gvk.String()))
+			}
+			return nil
+		})
+	}
+
+	// Any service ports using 443 are using http/http2 in their name (http not allowed on port 443)
+	for namespace := range meshNamespaces {
+		memberServices := &corev1.ServiceList{}
+		// listing for each member namespace, as we expect a large number of services in the whole cluster
+		if err := v.client.List(ctx, client.InNamespace(namespace), memberServices); err != nil {
+			return pkgerrors.Wrapf(err, "error listing Service resources in namespace %s", namespace)
+		}
+		for _, service := range memberServices.Items {
+			for _, port := range service.Spec.Ports {
+				if port.Port == 443 && (strings.HasPrefix(port.Name, "http-") || strings.HasPrefix(port.Name, "http2-")) {
```

Comment:
this doesn't catch the case where the port is simply named `http` or `http2` (which is what I used for testing, coincidentally)

Reply:
Thanks @dgn.  I'll update this and add a test case for it.